gluten-core/src/main/scala/org/apache/spark/sql/execution/python/EvalPythonExecTransformer.scala

```diff
@@ -0,0 +1,195 @@
+ * limitations under the License.
+ */
+package org.apache.spark.api.python
+
+import io.glutenproject.GlutenConfig
+import io.glutenproject.backendsapi.BackendsApiManager
+import io.glutenproject.execution.TransformContext
+import io.glutenproject.execution.TransformSupport
+import io.glutenproject.expression._
+import io.glutenproject.extension.GlutenPlan
+import io.glutenproject.metrics.MetricsUpdater
+import io.glutenproject.substrait.`type`._
+import io.glutenproject.substrait.SubstraitContext
+import io.glutenproject.substrait.expression._
+import io.glutenproject.substrait.extensions.ExtensionBuilder
+import io.glutenproject.substrait.plan.PlanBuilder
+import io.glutenproject.substrait.rel._
+
+import org.apache.spark.TaskContext
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions._
+import org.apache.spark.sql.execution.SparkPlan
+import org.apache.spark.sql.execution.python.EvalPythonExec
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.vectorized.ColumnarBatch
+
+import com.google.common.collect.Lists
+import com.google.protobuf.Any
+
+import java.util.ArrayList
+
+case class EvalPythonExecTransformer(
+    udfs: Seq[PythonUDF],
+    resultAttrs: Seq[Attribute],
+    child: SparkPlan)
+  extends EvalPythonExec
+  with TransformSupport
+  with GlutenPlan {
+
+  override def metricsUpdater(): MetricsUpdater =
+    BackendsApiManager.getMetricsApiInstance.genFilterTransformerMetricsUpdater(metrics)
+
+  override protected def evaluate(
+      funcs: Seq[ChainedPythonFunctions],
+      argOffsets: Array[Array[Int]],
+      iter: Iterator[InternalRow],
+      schema: StructType,
+      context: TaskContext): Iterator[InternalRow] = {
+    throw new NotImplementedError("EvalPythonExecTransformer doesn't support evaluate")
+  }
+
+  override protected def doExecute(): RDD[InternalRow] = {
+    throw new UnsupportedOperationException(s"EvalPythonExecTransformer doesn't support doExecute")
+  }
+
+  override protected def withNewChildInternal(newChild: SparkPlan): EvalPythonExecTransformer =
+    copy(udfs, resultAttrs, newChild)
+
+  override def columnarInputRDDs: Seq[RDD[ColumnarBatch]] = child match {
+    case c: TransformSupport =>
+      c.columnarInputRDDs
+    case _ =>
+      Seq(child.executeColumnar())
+  }
+
+  override def getBuildPlans: Seq[(SparkPlan, SparkPlan)] = child match {
+    case c: TransformSupport =>
+      val childPlans = c.getBuildPlans
+      childPlans :+ (this, null)
+    case _ =>
+      Seq((this, null))
+  }
+
+  override def getStreamedLeafPlan: SparkPlan = child match {
+    case c: TransformSupport =>
+      c.getStreamedLeafPlan
+    case _ =>
+      this
+  }
+
+  override def supportsColumnar: Boolean = GlutenConfig.getConf.enableColumnarIterator
+
+  override def doValidateInternal(): Boolean = {
+    if (BackendsApiManager.getBackendName.equalsIgnoreCase(GlutenConfig.GLUTEN_VELOX_BACKEND)) {
```

Comment:
I think we can simply define each backend's support state through interface `TransformerApi`.

Reply:
It is supported for both backends now.